sycl/test-e2e/Graph/RecordReplay/record_inconsistent_devices.cpp

```diff
@@ -0,0 +1,38 @@
+// REQUIRES: opencl-aot, level_zero, gpu
```

Comment:
if using `opencl` rather than `other-aot` works then i'd prefer that. `opencl` is the backend but `opencl-aot` if an ahead-of-time compilation tool.

Reply:
In my setup, It doesn't work if I use opencl instead of opencl-aot.
I put both in the updated version.

Reply:
How are you running this test? I'm struggling to get it to run, it always reports unsupported
```sh
ewan@devpc-1645:/media/ewan/big-ssd/dpcpp/build_all_release$ ./bin/sycl-ls
[opencl:cpu:0] Intel(R) OpenCL, Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz OpenCL 3.0 (Build 0) [2023.15.3.0.20_160000]
[opencl:cpu:1] Intel(R) OpenCL, Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz OpenCL 3.0 (Build 0) [2022.14.8.0.04_160000]
[opencl:acc:2] Intel(R) FPGA Emulation Platform for OpenCL(TM), Intel(R) FPGA Emulation Device OpenCL 1.2 [2023.15.3.0.20_160000]
[opencl:gpu:3] Intel(R) OpenCL HD Graphics, Intel(R) HD Graphics 530 [0x1912] OpenCL 3.0 NEO [22.07.22465]
[opencl:acc:4] Intel(R) FPGA Emulation Platform for OpenCL(TM), Intel(R) FPGA Emulation Device OpenCL 1.2 [2022.14.8.0.04_160000]
[ext_oneapi_level_zero:gpu:0] Intel(R) Level-Zero, Intel(R) HD Graphics 530 [0x1912] 1.3 [1.3.22465]

ewan@devpc-1645:/media/ewan/big-ssd/dpcpp/build_all_release$ ./bin/llvm-lit --param sycl_devices="ext_oneapi_level_zero:gpu;opencl:gpu" -sv ~/Development/dpcpp/sycl/test-e2e/Graph/RecordReplay/exception_inconsistent_devices.cpp -a
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:229: note: Targeted devices: ext_oneapi_level_zero:gpu, opencl:gpu
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:240: note: Running on multiple devices, XFAIL-marked tests will be skipped on corresponding devices
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:342: note: Found pre-installed AOT device compiler ocloc
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:345: warning: Couldn't find pre-installed AOT device compiler opencl-aot
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:361: note: Kernel fusion extension enabled
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:403: note: Aspects for ext_oneapi_level_zero:gpu: fp64, ext_intel_device_info_uuid, atomic64, ext_intel_gpu_eu_count, gpu, usm_device_allocations, online_compiler, ext_intel_gpu_eu_count_per_subslice, usm_shared_allocations, usm_host_allocations, ext_intel_memory_bus_width, ext_intel_legacy_image, ext_intel_gpu_slices, ext_intel_gpu_eu_simd_width, queue_profiling, ext_intel_gpu_hw_threads_per_eu, ext_intel_device_id, fp16, ext_intel_memory_clock_rate, online_linker, ext_intel_gpu_subslices_per_slice
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:412: note: SG sizes for ext_oneapi_level_zero:gpu: 16, 8, 32
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:403: note: Aspects for opencl:gpu: usm_host_allocations, usm_shared_allocations, ext_oneapi_srgb, ext_intel_legacy_image, queue_profiling, fp64, ext_intel_device_id, atomic64, fp16, gpu, usm_device_allocations, ext_intel_mem_channel, online_linker, online_compiler
llvm-lit: /home/ewan/Development/dpcpp/sycl/test-e2e/lit.cfg.py:412: note: SG sizes for opencl:gpu: 16, 8, 32
UNSUPPORTED: SYCL :: Graph/RecordReplay/exception_inconsistent_devices.cpp (1 of 1)
No supported devices to run the test on
********************

Testing Time: 0.01s
```

Reply:
I run it with using this command:
llvm-lit --param sycl_devices="ext_oneapi_level_zero:gpu;opencl:gpu" -sv sycl/test-e2e/Graph/RecordReplay/

Reply:
I can't reproduce this test running on my setup. I think it might actually be easier to remove the `REQUIRES` and just add a check that the backends we want are selected in the test, and skip if not.

```diff
iff --git a/sycl/test-e2e/Graph/RecordReplay/exception_inconsistent_devices.cpp b/sycl/test-e2e/Graph/RecordReplay/exception_inconsistent_devices.cpp
index 6bd71f887241..616323531a14 100644
--- a/sycl/test-e2e/Graph/RecordReplay/exception_inconsistent_devices.cpp
+++ b/sycl/test-e2e/Graph/RecordReplay/exception_inconsistent_devices.cpp
@@ -1,4 +1,3 @@
-// REQUIRES: opencl || opencl-aot, level_zero, gpu
 // RUN: %{build} -o %t.out
 // RUN: %{run-unfiltered-devices} %t.out
 //
@@ -24,6 +23,12 @@ int GetOpenclBackend(const sycl::device &Dev) {
 int main() {
   sycl::device Dev0{GetLZeroBackend};
   sycl::device Dev1{GetOpenclBackend};
+
+  if (Dev0.get_backend() != backend::ext_oneapi_level_zero ||
+      Dev1.get_backend() != backend::opencl) {
+    // Skip if we don't have two different devices
+    return 0;
+  }
   queue Queue{Dev0};

   exp_ext::command_graph Graph{Queue.get_context(), Dev1};
```

Reply:
I changed the test to make it more generic without specific device requirements